### PR TITLE
added TripChangeRequest

### PR DESCRIPTION
**Pull request by herlitze**
Added parts of TripChangeRequest as planned in issue #352 .

added:

Functionality:
Add WaitingTime at transfer (timed leg end/ timed leg start)

Parameter:
Single Trip to change (given by TR before)
TripLegRef
Flag for earlier departure or later arrival (only with StopPointRef)
TripRequestOptions (same structure as in TR)

not added yet:

Functionality:
Add sharing operator to an estimated trip

Parameter:
TripLegRef from and to or StopPointRef
Selected Operator

For the documentation: It is important to mention that sharing may change the whole trip structure (e.g. walk - bus - transfer - train may result in walk - sharing - transfer train)

### CODE REVIEW

**Comment by ue71603** [comment]
examples missing. Will write some.

File: OJP/OJP_Trips.xsd [hunk 1]
```diff
@@ -1699,4 +1699,78 @@
...
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="EarlierArrivalLaterDeparture" default="laterDeparture" minOccurs="0" maxOccurs="1">
+				<xs:annotation/>
+				<xs:simpleType>
+					<xs:restriction base="xs:string">
+						<xs:enumeration value="laterDeparture"/>
+						<xs:enumeration value="earlierArrival"/>
+					</xs:restriction>
+				</xs:simpleType>
```
Comment (ue71603) [inline OJP/OJP_Trips.xsd:1744]: Description of enum missing. Unclear what the parameter does.

File: OJP/OJP_Trips.xsd [hunk 2]
```diff
@@ -1699,4 +1699,78 @@
...
+			<xs:documentation>Trip change request parameter structure.</xs:documentation>
+		</xs:annotation>
+		<xs:sequence>
+			<xs:element name="ChangeLegRef" type="LegObjectIdType">
+				<xs:annotation>
+					<xs:documentation>Refers to the leg to be changed by the server.</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="EarlierArrivalLaterDeparture" default="laterDeparture" minOccurs="0" maxOccurs="1">
+				<xs:annotation/>
```
Comment (ue71603) [inline OJP/OJP_Trips.xsd:1738]: annotation missing

File: OJP/OJP_Trips.xsd [hunk 3]
```diff
@@ -1699,4 +1699,78 @@
...
+			<xs:group ref="TripContentFilterGroup"/>
+			<xs:element name="FareParam" type="FareParamStructure" minOccurs="0">
+				<xs:annotation>
+					<xs:documentation>Parameters for fare calculation. Only used if IncludeFare is set (TripContentFilterGroup).</xs:documentation>
+				</xs:annotation>
+			</xs:element>
+			<xs:element name="Extension" type="xs:anyType" minOccurs="0"/>
+		</xs:sequence>
+	</xs:complexType>
+	<xs:complexType name="EarlierArrivalLaterDepartureType"/>
```
Comment (ue71603) [inline OJP/OJP_Trips.xsd:1755]: empty element. What does it do.

**Review by ue71603 — CHANGES_REQUESTED** [review]

File: OJP/OJP_Trips.xsd (hunk 3 above)
Comment (herlitze) [reply OJP/OJP_Trips.xsd]: I fixed EarlierArrivalLaterDepartureType and added annotations.

**Comment by trurlurl** [comment]
Is there a reason not to have an *ErrorCondition*? To indicate that no alternative trip has be found, that *ChangeLegRef* does not refer to a valid *TransferLeg*, ..?

I don't remember the oral discussion in all detail - don't have the initial search parameters have to be known since part of the trip has to be recalculated?

**Comment by ue71603** [comment]
> Is there a reason not to have an _ErrorCondition_? To indicate that no alternative trip has be found, that _ChangeLegRef_ does not refer to a valid _TransferLeg_, ..?
> 
> I don't remember the oral discussion in all detail - don't have the initial search parameters have to be known since part of the trip has to be recalculated?

We should add one as well

File: OJP/OJP_Trips.xsd [hunk 4]
```diff
@@ -1746,6 +1738,11 @@
 				</xs:annotation>
 			</xs:element>
 			<xs:element name="Extension" type="xs:anyType" minOccurs="0"/>
+			<xs:element name="OperatorFilter" type="OperatorFilterStructure" minOccurs="0">
```
Comment (skinkie) [inline OJP/OJP_Trips.xsd:1741]: This should be above Extension.

File: examples/functions/change_trip/ChangeTripRequest.xml [hunk 5]
```diff
@@ -0,0 +1,211 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (herlitze) [inline examples/functions/change_trip/ChangeTripRequest.xml:1]: @ue71603 is this example still valid?

Comment (ue71603) [reply examples/functions/change_trip/ChangeTripRequest.xml:1]: changed.

**Comment by ue71603** [comment]
@herlitze  something very bad happened in this commit: https://github.com/VDVde/OJP/pull/357/commits/d183c5c5aefacfe0878d17996ae8601646e14b26  I will undo it, if possible

**Comment by ue71603** [comment]
did it now on: https://github.com/VDVde/OJP/pull/381 hope it works

**Pull request closed without merging**